### PR TITLE
Template Management for dynamic prompt

### PR DESCRIPTION
**Pull request by htamahc**
Merge after https://github.com/wso2/carbon-identity-framework/pull/1998

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package org.wso2.identity.integration.test.template.mgt;
```
Comment (pamodaaw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java:1]: License header is missing

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                "\"templateName\": \"" + templateName + "\", " +
+                "\"description\": \"" + description + "\"," +
+                "\"templateScript\": \"" + templateScript + "\"}";
+
+        String response = templateResource.contentType(MediaType.APPLICATION_JSON_TYPE).
+                accept(MediaType.APPLICATION_JSON_TYPE).post(String.class, addTemplateString);
+
+        return (JSONObject) JSONValue.parse(response);
+    }
+
```
Comment (pamodaaw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java:69]: remove additional new line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    @Test(alwaysRun = true, description = "Add template test")
+    public void testAddTemplate() {
+
+        String templateName = "Template1";
+        String description = "This is a template created by Alex";
+        String templateScript = "sample template script";
+        JSONObject response = addTemplate(templateName, description, templateScript);
+
```
Comment (pamodaaw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java:45]: remove additional new line

File: modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/template/mgt/TemplateManagementServiceClient.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import org.wso2.carbon.identity.template.mgt.model.TemplateInfo;
+import org.wso2.carbon.identity.template.mgt.ui.dto.TemplateRequestDTO;
+import org.wso2.carbon.identity.template.mgt.ui.dto.UpdateTemplateRequestDTO;
+import org.wso2.carbon.identity.template.mgt.ui.internal.TemplateManagementUIServiceDataHolder;
+import org.wso2.carbon.user.api.AuthorizationManager;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import java.util.List;
+
+public class TemplateManagementServiceClient {
```
Comment (pamodaaw) [inline modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/template/mgt/TemplateManagementServiceClient.java:36]: Add the class comment

File: modules/integration/tests-integration/pom.xml [hunk 5]
```diff
@@ -109,5 +109,9 @@
             <artifactId>org.wso2.carbon.um.ws.api</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.framework</groupId>
+            <artifactId>org.wso2.carbon.identity.template.mgt</artifactId>
+        </dependency>
     </dependencies>
 </project>
```
Comment (Abilashini) [inline modules/integration/tests-integration/pom.xml:117]: Add a blank line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java (hunk 1 above)
Comment (ChamathNS) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java:1]: fixed with https://github.com/wso2/product-is/pull/9220.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java (hunk 3 above)
Comment (ChamathNS) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java:45]: fixed with https://github.com/wso2/product-is/pull/9220.

File: modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/template/mgt/TemplateManagementServiceClient.java (hunk 4 above)
Comment (ChamathNS) [reply modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/template/mgt/TemplateManagementServiceClient.java:36]: fixed with https://github.com/wso2/product-is/pull/9220.

File: modules/integration/tests-integration/pom.xml (hunk 5 above)
Comment (ChamathNS) [reply modules/integration/tests-integration/pom.xml:117]: fixed with https://github.com/wso2/product-is/pull/9220.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java (hunk 2 above)
Comment (ChamathNS) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/template/mgt/TemplateManagementTestCase.java:69]: fixed with https://github.com/wso2/product-is/pull/9220.

**Pull request merged**